### ISSUE TITLE
Unknown Directive

### ISSUE THREAD

**Issue by mcrosson** [techknowlogick/caddy-s3browser#2]
When including this plugin as part of the Caddy download I can't seem to activate the plugin. I keep receiving an "Unknown Directive" error whenever I try to validate my Caddyfile. I've tried various combinations and configuration options with no success. I'm running minio on localhost at port 9009.

I've also checked ```caddy -plugins``` and http.s3browser is present.

Any help would be greatly appreciated.

My main caddy file:
```
# Individual configs are in their own files
import /etc/caddy/services/*.conf
```

My dl.website.conf file (where s3browser should be activating
```
dl.website.domain:80 dl.website.domain:443 {
    log /var/log/caddy/dl.log

    redir 301 {
        if {scheme} is http
        / https://dl.website.domain{uri}
    }

    s3browser {
        key [providedkey]
        secret [providedsecret]
        bucket [providedbucket]
        endpoint localhost
    }
    proxy / http://127.0.0.1:9009 {
        header_upstream Host 127.0.0.1
    }

    # Use acme.sh Let's Encrypt SSL cert setup
    tls fullchain.cer dl.website.domain.key
}
```

**Comment by techknowlogick** [comment]
First, thank you so much for trying out this caddy plugin. It makes me happy that someone else is even taking the time to look at it.

You've done everything that would be expected, and I too ran into the same problem as you when coding this, and I found out I needed to modify yet another file in caddy proper to have the directive be added in.

If you take a look at the Dockerfile, you'll see on [Line 9](https://github.com/techknowlogick/caddy-s3browser/blob/master/Dockerfile#L9) that I have to add "s3browser" into the list of directives that exist. I picked "cors" as the directive to go behind, but you probably could put it last in the list in `caddyhttp/httpserver/plugin.go`

Please let me know if that works, I will leave this issue open in case it doesn't.

**Comment by mcrosson** [comment]
@techknowlogick I saw that in the docker file but wasn't 100% what the end result was.

Does caddy provide any hooks for allowing directives to be added without having to tweak the sources?

I'm hoping to avoid custom builds of caddy for a project...

**Comment by techknowlogick** [comment]
I think all it requires is that I create a PR against caddy and add it in. I've just created https://github.com/mholt/caddy/pull/2383 and hopefully this'll help you avoid a custom build.

**Comment by techknowlogick** [comment]
@mcrosson alright, It has been merged. Fingers crossed things work out. I'll still leave this open just in case.

Sidenote: I've been casually following 🍭 ☁️ and I wish you luck with it.

**Comment by mcrosson** [comment]
@techknowlogick I saw it got merged :)

I'll be taking a closer look soon at updating caddy for the directive. I'm currently working on getting a number of docker images and more updated and don't want to rock the boat just yet ;) When I do the update, if it works I'll be sure to close out the ticket.

Thank you for the support for the 🍭 ☁️ ; I thought your username looked familiar but couldn't place it. I'm honestly shocked you connected it. ❤️
